### PR TITLE
SNXToken to implement ERC20

### LINKED ISSUE #290: The SNX module's satellite should implement the ERC20 standard

### LINKED ISSUE #296: SNXModule's ERC20 token implementation should not use namespace storage

### PR DESCRIPTION
**Pull request by leomassazza**
Fix #290
Fix #296

### CODE REVIEW

File: packages/synthetix-main/test/contracts/modules/SNXModule.test.js [hunk 1]
```diff
@@ -48,30 +50,56 @@ describe('SNXTokenModule', function () {
...
+          assert.equal(AnotherSNXToken.address, await NewSNX.getImplementation());
+        });
+
+        it('reads the upgraded SNX parameters', async () => {
+          assert.equal(await NewSNX.name(), 'Synthetix Network Token');
+          assert.equal(await NewSNX.symbol(), 'snx');
+          assert.equal(await NewSNX.decimals(), 18);
+        });
+
+        describe('New SNX can mint', () => {
```
Comment (leomassazza) [inline packages/synthetix-main/test/contracts/modules/SNXModule.test.js:92]: I'm having second thoughts about this test. I'd like to have it as a integration test / proof of concept, but every part is already covered.

@i-stam / @ajsantander  your thoughts?

File: packages/synthetix-main/contracts/mocks/token/SNXTokenMock.sol [hunk 2]
```diff
@@ -0,0 +1,20 @@
+//SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import "@synthetixio/core-contracts/contracts/proxy/UUPSImplementation.sol";
+import "@synthetixio/core-contracts/contracts/ownership/Ownable.sol";
+import "@synthetixio/core-contracts/contracts/token/ERC20.sol";
+
+contract SNXTokenMock is Ownable, UUPSImplementation, ERC20 {
```
Comment (leomassazza) [inline packages/synthetix-main/contracts/mocks/token/SNXTokenMock.sol:8]: See comment below. Needed for the test I added.

Comment (ajsantander) [reply packages/synthetix-main/contracts/mocks/token/SNXTokenMock.sol:8]: Shouldn't this extend SNXToken instead?

File: packages/synthetix-main/test/contracts/modules/SNXModule.test.js (hunk 1 above)
Comment (ajsantander) [reply packages/synthetix-main/test/contracts/modules/SNXModule.test.js:92]: Yeah, I suppose that this, like other tests, provide "peace of mind". It's not about coverage IMO, it's about proving a point. So I'm for keeping it here 👍

**Review by ajsantander — APPROVED** [review]
Looks good! Just left one minor comment.

**Review by i-stam — APPROVED** [review]

**Pull request merged**